cmd/akash/deployment.go

```diff
@@ -77,12 +105,77 @@ func doDeployCommand(ctx context.Context, cmd *cobra.Command, args []string) err
 	return nil
 }
 
-func parseDeployment(file string, tenant []byte, nonce uint64) (*types.Deployment, error) {
-	// todo: read and parse deployment yaml file
+func closeDeploymentCommand() *cobra.Command {
 
-	/* begin stub data */
-	deployment := testutil.Deployment(tenant, nonce)
-	/* end stub data */
+	cmd := &cobra.Command{
+		Use:   "close <deployment>",
+		Short: "close a deployment",
+		Args:  cobra.ExactArgs(1),
+		RunE: context.WithContext(
+			context.RequireKey(context.RequireNode(closeDeployment))),
+	}
 
-	return deployment, nil
+	context.AddFlagNode(cmd, cmd.Flags())
+	context.AddFlagKey(cmd, cmd.Flags())
+	context.AddFlagNonce(cmd, cmd.Flags())
+
+	return cmd
+}
+
+func closeDeployment(ctx context.Context, cmd *cobra.Command, args []string) error {
+	kmgr, err := ctx.KeyManager()
+	if err != nil {
+		return err
+	}
+
+	key, err := ctx.Key()
+	if err != nil {
+		return err
+	}
+
+	nonce, err := ctx.Nonce()
+	if err != nil {
+		return err
+	}
+
+	signer := txutil.NewKeystoreSigner(kmgr, key.Name, constants.Password)
```

Comment:
ot: i think a `Signer() (txtutil.Signer,keys.Info,error)` method on `Context` would save a lot of boilerplate.

Reply:
opened #131